src/Menu/Provider/GroupMenuProvider.php

```diff
@@ -170,4 +121,44 @@ public function has($name, array $options = [])
     {
         return 'sonata_group_menu' === $name;
     }
+
+    private function generateMenuItem(array $item, array $group)
+    {
+        if (isset($item['admin']) && !empty($item['admin'])) {
+            $admin = $this->pool->getInstance($item['admin']);
+
+            // skip menu item if no `list` url is available or user doesn't have the LIST access rights
+            if (!$admin->hasRoute('list') || !$admin->hasAccess('list')) {
+                return null;
+            }
+
+            $label = $admin->getLabel();
+            $options = $admin->generateMenuUrl('list', [], $item['route_absolute']);
+            $options['extras'] = [
+                'label_catalogue' => $admin->getTranslationDomain(),
+                'admin' => $admin,
+            ];
+        } else {
```

Comment:
Please avoid this `else` buy using an early return statement at the end of the `if` block.

Reply:
I can do that, but then I will have to duplicate the line:

`return $this->menuFactory->createItem($label, $options);`

are you ok with it?